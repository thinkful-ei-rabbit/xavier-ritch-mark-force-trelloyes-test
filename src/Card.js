import React, { Component } from 'react';
import Store from './STORE'
import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      deleteClicked: false
     }

     const handleDeleteCard = (id) => {
      Store.filter((item) => item.id !== id);
      this.setState({
        deleteClicked: true
      })
     }
  }

  render() { 
    return ( 
      <div className='Card'>
      <button
        type='button'
        onClick={this.props.onDeleteCard}
      >
        delete
      </button>
      <h3>{this.props.title}</h3>
      <p>{this.props.content}</p>
    </div>
     );
  }
}
 
export default Card;
