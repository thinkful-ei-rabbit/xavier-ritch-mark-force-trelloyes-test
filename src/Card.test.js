import React from "react";
import ReactDOM from "react-dom";
import Card from "./Card";
import renderer from 'react-test-renderer';



describe("<Card />", () => {
  it("renders Card without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<Card 
        title="Test Card"
        content="this is a test card"
    />, div);

    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders UI", () => {
    const tree = renderer
        .create(<Card 
            title="Test Card 2"
            content="this is another test card"
        />).toJSON();
        expect(tree).toMatchSnapshot();
  });

});


