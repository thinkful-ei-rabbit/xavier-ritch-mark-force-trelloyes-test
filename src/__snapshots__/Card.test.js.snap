// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders UI 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Test Card 2
  </h3>
  <p>
    this is another test card
  </p>
</div>
`;
